GSOC18/GeometryUtilities-master/interpolation/test_coef.py

```diff
@@ -0,0 +1,138 @@
+#Importing custom classes and function
+from sq_Cells import sq_Cells
+from scipy import misc
+
+# Parameters ... to fix later
+#input_default_file = '/data_CMS/cms/grasseau/HAhRD/test_triggergeom.root'
+input_default_file='geometry_data/test_triggergeom.root'
+coef_filename='sq_cells_data/coef_dict_layer_1_len_0.7.pkl'
+#This need to be manually entered now from the screen output in main script
+resolution = (442,442)
+
+################ DRIVER FUNCTION DEFINITION ###################
+def readGeometry( input_file,  layer, subdet ):
+    '''
+    AUTHOR: Grasseau Gilles
+    DESCRIPTION:
+        This function reads the root file which contain the Geometry
+    of the detector and create a dictionary of "Cell" object assiciated
+    with every hexagonal cell in the detector.
+    USAGE:
+        INPUT:
+            input_file  : the name of input geometry file (root file)
+            Layer       : which layer's cell we are interested in
+            Subdet      : which part of subdetector it is
+                            (EE,...)
+        OUTPUT:
+            cells_d     : the hexagonal cell-dictionary with id of
+                          cell as the key and Cell object as value
+    '''
+
+    t0 = datetime.datetime.now()
+    treename = 'hgcaltriggergeomtester/TreeCells'
+    cells = read_geometry(filename=input_file, treename=treename,
+              subdet=subdet, layer=layer, wafer=-1)
+    cells_d = dict([(c.id, c) for c in cells])
+    t1 = datetime.datetime.now()
+    print 'Cells read: number=', len(cells), ', time=', t1-t0
+    return cells_d
+
+# Compute and copare the total areas (hex, square)
+def compareAreas( cells_d, sq_coef):
+    layerArea = float(0.0)
+    for k in cells_d.keys():
+        layerArea =  layerArea + cells_d[k].vertices.area
+
+    squareArea = float( 0.0 )
+    for cell_id in sq_coef.keys():
+        area = cells_d[ cell_id ].vertices.area
+        for sq in sq_coef[ cell_id ]:
+            squareArea = squareArea + sq[1]*area
+
+    print "Layer  Area :", layerArea
+    print "Square Area :", squareArea
+
+    if ( abs( (layerArea - squareArea)/layerArea ) < 1.e-07 ):
+        print "Surface test PASS"
+    else:
+        print "Surface test ERROR"
+
+    return
+
+# Map the cell coef to a regular grid
+def mappingOnMatrix( cells_d, sq_coef, resolution ):
+
+    # Spread the coeficient in the squared grid
+    sCells = np.zeros((resolution[0], resolution[1]) )
+    for cell_id in sq_coef.keys():
+        area = cells_d[ cell_id].vertices.area
+        for sq in sq_coef[ cell_id ]:
+            i,j =  sq[0]
+            sCells[i][j] = sCells[i][j] + sq[1]*area
+
+    return sCells
+
+
+# Plot the image ..
+def plotImage( sCells ):
+
+    iSize, jSize = sCells.shape
+    sMax = sCells.max()
+    print 'The maximum value is: ',sMax
+
+    ima = np.zeros( (iSize, jSize, 3), dtype=np.uint8)
+    for i in range(iSize):
+        for j in range(jSize):
+            val = int( sCells[i][j] / sMax * 255)

```

Comment:
Here, **int** is making the effective functionality of **floor function**. Since all the value will be less than the maximum one, the ratio will be less than 1 between sCells[i][j] and val.
That's the reason I have made 
`if val>=254 || val>0`
So, accommodate that flooring effect of int.